Community/Tdarr_Plugin_Soggys_NVENC_HEVC_CQV_Optimised_Bitrate.js

```diff
@@ -0,0 +1,220 @@
+
+      \\nExample:\\n
+      3`
+      },
+      {
+        name: 'ffmpeg_preset',
+        type: 'string',
+        defaultValue: 'medium',
+        inputUI: {
+          type: 'text',
+        },
+        tooltip: `OPTIONAL, DEFAULTS TO MEDIUM IF NOT SET
+      \\n Enter the ffmpeg preset you want, leave blank for default (medium)
+
+      \\nExample:\\n
+        slow
+
+      \\nExample:\\n
+        medium
+
+      \\nExample:\\n
+        fast
+
+      \\nExample:\\n
+        veryfast`
+      }
+    ]
+  }
+}
+
+// eslint-disable-next-line no-unused-vars
+const plugin = (file, librarySettings, inputs, otherArguments) => {
+
+    const lib = require('../methods/lib')();
+  // eslint-disable-next-line no-unused-vars,no-param-reassign
+  inputs = lib.loadDefaultValues(inputs, details);
+  var subcli = `-c:s copy`
+  var maxmux = ''
+  var map = '-map 0'
+  //default values that will be returned
+  var response = {
+    processFile: false,
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: false,
+    reQueueAfter: true,
+    infoLog: '',
+  }
+
+  const streamHeight = file.ffProbeData.streams[0].height;
+  const streamWidth = file.ffProbeData.streams[0].width;
+  const streamFPS = file.mediaInfo.track[0].FrameRate;
+  let streamBR = file.bit_rate;
+
+  //check if the file is a video, if not the function will be stopped immediately
+  if (file.fileMedium !== 'video') {
+    response.processFile = false
+    response.infoLog += '☒File is not a video! \n'
+    return response
+  } else {
+    // bitrateprobe = file.ffProbeData.streams[0].bit_rate;
+    response.infoLog += '☑File is a video! \n'
+  }
+
+  optimalVideoBitrate = Math.floor((streamHeight * streamWidth * streamFPS) * inputs.targetCodecCompression);
+  response.infoLog += 'Optimal Bitrate = ' + (optimalVideoBitrate / 1000000).toString() + '\n'
+  response.infoLog += 'Stream Bitrate = ' + (streamBR / 1000000).toString() + '\n'
+  if (file.ffProbeData.streams[0].codec_name == 'hevc'
+  && (isNaN(optimalVideoBitrate) || isNaN(streamBR) || streamBR < optimalVideoBitrate)) {
+    response.processFile = false
+    response.infoLog += '☑File is already in hevc and is below optimal bitrate!\n'
+    return response
+  }
+  else if (file.ffProbeData.streams[0].codec_name == 'hevc'
+  && (file.video_resolution === '480p' || file.video_resolution === '576p' || file.video_resolution === '720p')) {
+    response.processFile = false
+    response.infoLog += '☑File is low res, wont transcode!\n'
+    return response
+  }
+
+  // Check if preset is configured, default to slow if not
+  var ffmpeg_preset
+  if (inputs.ffmpeg_preset === undefined) {
+    ffmpeg_preset = `slow`
+    response.infoLog += '☑Preset not set, defaulting to slow\n'
+  } else {
+    ffmpeg_preset = `${inputs.ffmpeg_preset}`
+    response.infoLog += `☑Preset set as ${inputs.ffmpeg_preset}\n`
+  }
```

Comment:
This will never be undefined. Better to do `inputs.ffmpeg_preset` since that will also check if it's empty.